package_metrics/parsing_utils.py

```diff
@@ -0,0 +1,13 @@
+def behind(latest, current):
+    delta = [0, 0, 0]
+    for index, (lat, cur) in enumerate(zip(vsplit(latest), vsplit(current))):
+        if lat != cur:
+            delta[index] = lat - cur
+            break
+    return delta
+
+
+def vsplit(version):
+    padded = f"{version}.0.0"  # append extra zeros for versions like '1' or '2.0'
+    return [int(n) for n in padded.split(".")[:3]]
```

Comment:
My understanding is that with pip it is possible to get something like `2.0rc1`, or even something like `alpha`, so it definitely is worth handling this. As far as how, we could either ignore exceptions raised when parsing version identifiers (basically not track these packages in any way), or track these packages as a separate category of "unknown". I'm leaning towards the new "unknown" category, but would be curious to hear your thoughts.

Related, Joel had added the concept of [exotic](https://github.com/dimagi/commcare-hq/blob/8fe9a7efe20233c2732a473521fe5619ca70fe30/scripts/yarn-list.py#L57) for JS specifically as it seems there was a possibility of no latest version being found when parsing JS packages, but I'd be tempted to lump all of that into the "unknown" category as well if we go that direction.

Reply:
Adding an "unknown" category seems reasonable.

Another probably easier option is to just default to `0` if the value cannot be converted to an integer since most augmentations like _alpha_, _beta_, and _rc_ are pre-0. Could that cause incorrect output in some cases? Is that a problem?

Reply:
Hmm interesting. That would probably be fine for most cases because of what you mentioned, but there is still the potential for negative deltas. For example if 2.1 is the version we use, but 2.2rc1 exists (this seems unlikely, but possible), then the delta would show up at 0.-1.0 behind the latest version 🤔 I suppose that could be resolved by parsing out the first number in the string `2rc1` and using that instead of 0, but wouldn't want to attempt to accommodate this any more than that.

Reply:
Parsing out the first number seems good. If it that doesn't yield a number then classify as "unknown".